### PR TITLE
Built-In Transform support for Apple Log

### PR DESCRIPTION
**Pull request by JGoldstone**
Address request for Apple Log support as raised in #1916

### CODE REVIEW

**Review by doug-walker — CHANGES_REQUESTED** [review]
Looks great, thank you Joseph!

One more piece needs to be added. There is a check that is done in Config.cpp to ensure that a given version of the config format does not use Builtin functions that are for a newer version.  Apple Log will be the first builtin added for config format 2.4, so please add a new section [here](https://github.com/AcademySoftwareFoundation/OpenColorIO/blob/0c90ded16687b1b63f3e24ef417789c89b1dc509/src/OpenColorIO/Config.cpp#L5234) that tests if the new builtin is being used with older formats.

File: src/OpenColorIO/transforms/builtins/AppleCameras.cpp [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    constexpr double R_t   = 0.01;
+    constexpr double c     = 47.28711236;
+    constexpr double beta  = 0.00964052;
+    constexpr double gamma = 0.08550479;
+    constexpr double delta = 0.69336945;
+    const double P_t       = c * pow((R_t - R_0), 2.0);
+    
+    if (in >= P_t)
+    {
+        return pow(2.0, (in - delta) / gamma) - beta;
```
[the remark this answers is not in the document]
Comment (JGoldstone) [reply src/OpenColorIO/transforms/builtins/AppleCameras.cpp]: I believe these have been addressed; please review.

File: src/OpenColorIO/transforms/builtins/AppleCameras.cpp [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+namespace APPLE
+{
+
+void RegisterAll(BuiltinTransformRegistryImpl & registry) noexcept
+{
+    {
+        auto APPLE_LOG_to_ACES2065_1_Functor = [](OpRcPtrVec & ops)
+        {
+            CreateLut(ops, 4096, APPLE_LOG::GenerateLutValues);
```
[the remark this answers is not in the document]
Comment (JGoldstone) [reply src/OpenColorIO/transforms/builtins/AppleCameras.cpp]: I have tried to mimic that code as closely as I can, but what I have produced segfaults. Is there some additional work I should be doing based on context I'm not understanding here? Your assistance would be very much appreciated.

Oh, and I added that check for config format 2.4.

[the remark this answers is not in the document]
Comment (JGoldstone) [reply src/OpenColorIO/transforms/builtins/AppleCameras.cpp]: Yes, it was unanticipated context. As you pointed out, this is the first builtin added that requires 2.4, but the config stored in the code as a literal to enable testing was still (reasonably enough) at 2.3. Similarly the last supported minor version for OCIO 2.x needs now to be 4, not 3. CI will tell me soon enough if I've overlooked anything else.

**Review by doug-walker — APPROVED** [review]
LGTM!

**Review by remia — APPROVED** [review]

**Comment by JGoldstone** [comment]
Thanks, both of you, for the speedy review. Not having write access myself, could I ask one of you to merge the request and close this out?

**Comment by carolalynn22** [comment]
Let’s wait for Robert to take a look, please. I’ve let him know it’s up.

**Comment by KelSolaar** [comment]
This would be a good PR to add as a developer guide example, i.e., how to add a new transformation.

**Comment by JGoldstone** [comment]
To Thomas' point, yes, including the introspective aspects, _i.e._ not allowing the new builtins to be used in pre-3.4 configs, and being the first thing in a release to have such a constraint.

**Review by zachlewis — APPROVED** [review]

**Comment by rkmolholm** [comment]
Looks good to me - thank you Joseph!

**Comment by doug-walker** [comment]
@rkmolholm , after looking at the results, I'm not sure this transform is correct.  It may simply be that the footage I have was not correctly captured.  Could you point us at some known good test material in Apple Log space?

**Pull request merged**